autosubmitconfigparser/config/configcommon.py

```diff
@@ -1837,12 +1838,31 @@ def reload(self, force_load=False, only_experiment_data=False, save=False):
             self.experiment_data = self.deep_read_loops(self.experiment_data)
             self.experiment_data = self.substitute_dynamic_variables(self.experiment_data)
             self.experiment_data = self.substitute_dynamic_variables(self.experiment_data, in_the_end=True)
+            if not self.experiment_data.get("AUTOSUBMIT", {}):  # Reserved namespace for autosubmit
+                self.experiment_data["AUTOSUBMIT"] = {}
+            else:
+                Log.warning("AUTOSUBMIT namespace is reserved for Autosubmit, please don't use it in your configuration")
             self.misc_data = {}
             self.misc_files = list(set(self.misc_files))
             for filename in self.misc_files:
                 self.misc_data = self.unify_conf(self.misc_data,
                                                  self.load_config_file(self.misc_data, Path(filename), load_misc=True))
             self.load_current_hpcarch_parameters()
+            self.load_workflow_commit()
+
+    def load_workflow_commit(self) -> None:
+        """
+        Load the workflow commit from the .git folder
+        """
+        self.experiment_data["AUTOSUBMIT"]["WORKFLOW_COMMIT"] = "not-versioned"
```

Comment:
Maybe it'd be better to leave it empty, and fill the `WORKFLOW_COMMIT` only when there is a commit? This way users won't have to ever worry about handling `if %AUTOSUBMIT.WORKFLOW_COMMIT% == "not-versioned"`... same for RO-Crate, this way we wouldn't risk having a commit being saved somewhere as "not-versioned".

Reply:
We can leave it null in the DB. I am thinking in the provenance archive, and at least in rocrate it'd be easier if it were empty instead of having a string. Otherwisem I'd need to handle that before using the value.